src/ir/context.rs

```diff
@@ -1910,6 +1928,22 @@ impl<'ctx> BindgenContext<'ctx> {
         // type parameter in array or not.
         self.has_type_param_in_array.as_ref().unwrap().contains(id)
     }
+
+    /// Compute whether the type has float.
+    fn compute_has_float(&mut self) {
+        assert!(self.has_float.is_none());
+        self.has_float = Some(analyze::<HasFloat>(self));
```

Comment:
This should be conditionally computed depending on whichever options enable deriving something that depends on this information. Right now that is only `self.options.derive_eq`, right?